R/db_manipulating_functions.R

```diff
@@ -28,31 +28,56 @@
 #' @importFrom rlang .data
 #' @export
 filter_keys <- function(.data, filters, by = NULL, na_by = NULL) {
-
-  # Check arguments
-  assert_data_like(.data)
-  assert_data_like(filters, null.ok = TRUE)
-  checkmate::assert_character(by, null.ok = TRUE)
-  checkmate::assert_character(na_by, null.ok = TRUE)
-
   if (is.null(filters)) {
     return(.data)
-  } else {
-    if (is.null(by) && is.null(na_by)) {
-      # Determine key types
-      key_types <- filters |>
-        dplyr::ungroup() |>
-        dplyr::summarise(dplyr::across(.cols = tidyselect::everything(), .fns = ~ any(is.na(.), na.rm = TRUE))) |>
-        tidyr::pivot_longer(tidyselect::everything(), names_to = "column_name", values_to = "is_na")
-
-      by    <- key_types |> dplyr::filter(!.data$is_na) |> dplyr::pull("column_name")
-      na_by <- key_types |> dplyr::filter(.data$is_na)  |> dplyr::pull("column_name")
-
-      if (length(by) == 0)    by    <- NULL
-      if (length(na_by) == 0) na_by <- NULL
-    }
-    return(inner_join(.data, filters, by = by, na_by = na_by))
   }
```

Comment:
`assert_character(by)` most likely disappeared when I removed `assert_character(na_by)` due to `na_by` no longer being defined in the function signature.

I reinstated the checks, but since it is possible to skip to [the `dplyr::inner_join` call](https://github.com/ssi-dk/SCDB/blob/ffaf1b184fc241726df8e2f354c09ea0c3482a71/R/db_manipulating_functions.R#L59), but have opted for `checkmate::check_subset` instead of `check_set_equal` (which would require the user to specify **all** columns in `by` or `na_by` if **any** were given).